rcl/src/rcl/arguments.c

```diff
@@ -1100,6 +1100,15 @@ rcl_arguments_copy(
       }
     }
   }
+  char * security_context_copy = rcutils_strdup(args->impl->security_context, allocator);
+  if (args->impl->security_context && !security_context_copy) {
+    if (RCL_RET_OK != rcl_arguments_fini(args_out)) {
+      RCL_SET_ERROR_MSG("Error while finalizing arguments due to another error");
+    }
+    RCL_SET_ERROR_MSG("Error while copying security context argument");
```

Comment:
@ivanpauno won't you get a complain about the error message being overwritten if `rcl_arguments_fini` fails?

Reply:
Done in https://github.com/ros2/rcl/pull/515/commits/77aad14964b484c6091bebe0cec28169546303a9.